src/bindings/js/node/src/core_wrap.cpp

```diff
@@ -27,7 +28,8 @@ Napi::Object CoreWrap::Init(Napi::Env env, Napi::Object exports) {
 
     Napi::FunctionReference* constructor = new Napi::FunctionReference();
     *constructor = Napi::Persistent(func);
-    env.SetInstanceData(constructor);
+    auto data = env.GetInstanceData<AddonData>();
```

Comment:
Is it certain that `GetInstanceData` returns valid `AddonData` reference/object?

Reply:
Those methods are not exposed to js. As long as we use `SetInstanceData` once in `addon.cpp InitAll()` it should be a valid pointer